gym_env.py

```diff
@@ -0,0 +1,209 @@
+import warnings
+from typing import Tuple, Optional
+
+from cv2 import resize, INTER_AREA, INTER_LINEAR
+import gym
+from gym import spaces
+from gym.core import ObsType, ActType
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+from .. import utils
+
+
+class PendulumEnv(gym.Env):
+
+    _gravity_const = 9.81
+    
+
+    def __init__(self, exploring_starts: bool = True, start_position: Tuple = None,
+                 terminate_on_goal: bool = True,
+                 hidden_goal: bool = False,
+                 init_state: Tuple = None,
+                 should_render: bool = True):
+    
+        #TODO: make sure should_render implemented properly (used be initialize_env) so that set_rendering can work
+
+        """
+        Pendulum Environment: Gym Wrapper
+
+        Notes:
+            removed fixed_goal argument since goal always fixed in pendulum
+
+            dimensions now represents the render size
+        """
+
+        self.environment = gym.make('Pendulum-v1', g= self._gravity_const)
+
+        self.exploring_starts = exploring_starts if start_position is None else False
+        self.fixed_goal = True
+        self.hidden_goal = hidden_goal
+        self.terminate_on_goal = terminate_on_goal
+        self.should_render = should_render
+        
+
+        self.action_space = self.environment.action_space
+        self.goal_state = np.array([0.0, 0.0, 0.0])
+
+        self._initialize_state(init_state)
+        self._initialize_state_space()
+        self._initialize_obs_space()
+
+    
+    def _initialize_state_space(self):
+        
+        #state represented as: (x, y, angular vel) with ranges ([-1,1], [-1,1], [-8,8])
+        #TODO: maybe use a Dict representation with tags for each box in state?
+        if self.hidden_goal:
```

Comment:
Remove check and remove `self.hidden_goal` variable

Reply:
Done